opentelemetry-api/src/opentelemetry/_logs/_internal/__init__.py

```diff
@@ -57,8 +58,23 @@ class LogRecord(ABC):
     pertinent to the event being logged.
     """
 
+    @overload
     def __init__(
         self,
+        *,
+        timestamp: Optional[int] = None,
+        observed_timestamp: Optional[int] = None,
+        context: Optional[Context] = None,
+        severity_text: Optional[str] = None,
+        severity_number: Optional[SeverityNumber] = None,
+        body: AnyValue = None,
+        attributes: Optional[_ExtendedAttributes] = None,
+    ) -> None: ...
+
+    @overload
+    def __init__(
+        self,
+        *,
```

Comment:
Not that I don't like it but making it require keyword-only arguments isn't a breaking change?